rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/RpkiRepositoryValidationService.java

```diff
@@ -152,12 +149,12 @@ public void validateRrdpRpkiRepository(long rpkiRepositoryId) {
         });
 
         boolean triggerCaTreeAfter = false;
-        boolean changedAtLeastOneObject = false;
         try {
             final String uri = rpkiRepository.getRrdpNotifyUri();
             if (isRrdpUri(uri)) {
-                changedAtLeastOneObject = rrdpService.storeRepository(rpkiRepository, validationRun);
+                boolean changedAtLeastOneObject = rrdpService.storeRepository(rpkiRepository, validationRun);
 
+                triggerCaTreeAfter = changedAtLeastOneObject || rpkiRepository.isPending();
```

Comment:
Can you clarify on this one? Why do we run tree valdiation in case of `rpkiRepository.isPending()`?

Reply:
Certificate tree validation skips any objects from a pending repository, so the output is different when a repository is pending. So when the repository switches to failed or downloaded the certificate tree validation needs to run again.

Before this PR this was usually not a problem since all pending repositories would be new and at least one object would be downloaded. But now that repositories become pending on startup it is often the case that no changes are detected since the repository may have been downloaded (e.g. when the server is restarted there is almost no time between the last download and startup).